### ISSUE TITLE
Further optimize SpriteSolidColor by re-using textures for rectangles of the same color

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#975]
### What should be added/changed?

**tl;dr** Make all `SpriteSolidColor`s of the same color use the same texture but stretched. no need to waste memory or time generating new textures.

This is not a high priority ticket, but it may be useful, especially as part of an effort to expand or refactor shape texture generation.

I [asked (discord link)](https://discord.com/channels/458662222697070613/458662458198982676/892986995708678144) @einarf if we could further optimize SpriteSolidColor beyond the scope of #974 by re-using and stretching the same texture for all `SpriteSolidColor`s of the same color. He said that it might be worthwhile, but would require the textures be at least 32px by 32px to prevent artifacts when zooming in and out. 

### What would it help with?

Reduce memory use and increase performance even further when making many `SpriteSolidColor`s instances. This would be useful in a number of scenarios. One example is cases like gamejams and early prototyping where terrain and objects may all be rectangles of the same color.

Another usecase is giving students more enjoyable and impactful experiences in educational environments where hardware may be underpowered. For a more concrete user story, imagine that beginner students have an assignment to make a retro game. They choose to use a volcano as a set piece, and are using `SpriteSolidColor` as a lava particle. It's likely that students will try a very high number of lava particles at some point. As beginners, the students will not understand how to perform any advanced optimizations such as shaders to display particles. Optimizing `SpriteSolidColor` as described earlier may allow the students to watch a very large number of chunks of lava of different sizes all smoothly fly out of the volcano at once. This sort of experience is highly rewarding to younger students.

**Comment by einarf** [comment]
This was merged into the previous release.
